src/Roslyn.Diagnostics.Analyzers/CSharp/CSharpAvoidOptSuffixForNullableEnableCode.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Analyzer.Utilities.Extensions;
+using Analyzer.Utilities.Lightup;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Roslyn.Diagnostics.Analyzers;
+
+namespace Roslyn.Diagnostics.CSharp.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class CSharpAvoidOptSuffixForNullableEnableCode : AvoidOptSuffixForNullableEnableCode
+    {
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterSyntaxNodeAction(context =>
+            {
+                var parameter = (ParameterSyntax)context.Node;
+
+                if (parameter.Identifier.Text.EndsWith("Opt", System.StringComparison.Ordinal) &&
+                    context.SemanticModel.GetNullableContext(parameter.SpanStart).AnnotationsEnabled())
+                {
+                    context.ReportDiagnostic(parameter.Identifier.CreateDiagnostic(Rule));
+                }
+            }, SyntaxKind.Parameter);
+
+            context.RegisterSyntaxNodeAction(context =>
+            {
+                var fieldDeclaration = (FieldDeclarationSyntax)context.Node;
+
+                foreach (var variable in fieldDeclaration.Declaration.Variables)
+                {
+                    if (variable.Identifier.Text.EndsWith("Opt", System.StringComparison.Ordinal) &&
+                        context.SemanticModel.GetNullableContext(variable.SpanStart).AnnotationsEnabled())
```

Comment:
In theory it's possible to have some parts of a field declaration after the `#nullable enable` but I don't think that would happen in Roslyn so we might want to check the context at the field declaration level rather than each variable.